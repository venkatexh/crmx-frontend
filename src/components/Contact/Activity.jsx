import {Link} from "react-router-dom";

const Activity = ({activity}) => {
  const {type, text, doneAt, refId, campaignName} = activity;

  const iconType = () => {
    if (type === 'Added') {
      return 'plus';
    } else if (type === 'Campaign') {
      return 'email';
    } else if (type === 'Subscribed') {
      return 'subscribe';
    } else if (type === 'Unsubscribed') {
      return 'unsubscribe';
    }
  }

  const renderActivity = () => {
    if (type === 'Campaign') {
      return (
        <div className={'activityCardLeft'}>
          <img src={`/icons/${iconType()}.png`} className={'activityIcon'} alt={'icon'}/>
          <div>Contact was sent campaign&nbsp;</div>
          <Link to={`/campaign/${refId}`}>{campaignName}</Link>
        </div>
      );
    } else if (type === 'Added') {
      return (
        <div className={'activityCardLeft'}>
          <img src={`/icons/${iconType()}.png`} className={'activityIcon'} alt={'icon'}/>
          <div>{text}</div>
        </div>
      );
    } else if (type === 'Subscribed') {
      return (
        <div className={'activityCardLeft'}>
          <img src={`/icons/${iconType()}.png`} className={'activityIcon'} alt={'icon'}/>
          <div>Contact was subscribed</div>
        </div>
      )
    } else if (type === 'Unsubscribed') {
      return (
        <div className={'activityCardLeft'}>
          <img src={`/icons/${iconType()}.png`} className={'activityIcon'} alt={'icon'}/>
          <div>Contact was unsubscribed</div>
        </div>
      )
    }
  }
  return <>
    <div className={'activityCard'}>
      {
        renderActivity()
      }
      <div className={'activityCardMiddle'}>
        <img src={'/icons/long-arrow.png'} className={'longArrow'} alt={'icon'}/>
      </div>
      <div className={'activityCardRight'}>{doneAt}</div>
    </div>
    <hr/>
  </>
}

export default Activity;