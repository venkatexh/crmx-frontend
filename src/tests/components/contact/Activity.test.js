import { shallow } from "enzyme";
import React from "react";
import Activity from "../../../components/Contact/Activity";

it("Contact activity component snapshot", () => {
  expect(
    shallow(
      <Activity
        activity={{
          type: "test type",
          text: "updated",
          doneAt: "12 PM",
          refId: "testid123",
          campaignName: "test campaign",
        }}
      />
    )
  ).toMatchSnapshot();
});
