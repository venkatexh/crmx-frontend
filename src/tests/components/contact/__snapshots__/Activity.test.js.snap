// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact activity component snapshot 1`] = `ShallowWrapper {}`;
